### PR TITLE
Use Bi-LSTM model for Spouse Demo

### PR DESCRIPTION
**Pull request by brahmaneya**
## Description of proposed changes
Change spouse demo final model to use a bidirectional LSTM. 

## Related issue(s)
https://github.com/snorkel-team/snorkel-tutorials/issues/147

Fixes # (issue)
https://github.com/snorkel-team/snorkel-tutorials/issues/147

## Test plan
tox

## Checklist

Need help on these? Just ask!

* [x] I have read the **CONTRIBUTING** document.
* [x] I have verified that my changes are covered by continuous integration.
* [x] All new and existing tests passed.

### CODE REVIEW

File: spouse/tf_model.py [hunk 1]
```diff
@@ -2,56 +2,85 @@
...
+from tensorflow.keras.layers import (
+    Bidirectional,
+    Concatenate,
+    Dense,
+    Embedding,
+    Input,
+    LSTM,
+)
+
+
```
Comment (paroma) [inline spouse/tf_model.py]: I thought there’s preprocessors for getting left and right tokens, any way to reuse them?

Comment (paroma) [inline spouse/tf_model.py]: Is there any way to reuse the preprocessors that get the left, right and between tokens? 

https://github.com/snorkel-team/snorkel-tutorials/blob/b6f85b89e623e2dd00ca23bf2572390fd7dd0b80/spouse/preprocessors.py#L55

Comment (paroma) [reply spouse/tf_model.py]: And is person1 always to the left of person2? If so, then the min() shouldn’t be needed

Comment (brahmaneya) [reply spouse/tf_model.py]: Switched to using existing fields for right tokens and between tokens (which were already present). For left tokens, it's just one line, so I inlined it.

File: spouse/tf_model.py [hunk 2]
```diff
@@ -2,56 +2,68 @@
...
+    Embedding,
+    Input,
+    LSTM,
+)
 
 
 def get_feature_arrays(df: pd.DataFrame) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
     """Get np arrays of upto max_length tokens and person idxs."""
+    bet = df.between_tokens
+    left = df.apply(lambda c: c.tokens[: c.person1_word_idx[0]][-4:-1], axis=1)
```
Comment (paroma) [inline spouse/tf_model.py]: this would only pick up the last 3 tokens, not all of them right?

Comment (brahmaneya) [reply spouse/tf_model.py]: Yes, that is intentional (the existing right_tokens field also does that).

**Pull request merged**